app/models/user.rb

```diff
@@ -22,6 +22,8 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+         
+  before_validation :sanitize_member_photo_basename, if: -> { member_photo_file_name.present? && member_photo_file_name_changed? }
```

Comment:
I am thinking we may want to do this as an `after_validation` action.

The reasons are
1. `sanitize_member_photo_basename` is, in effect, validating the file name, _before_ paperclip will actually do the exact same thing,
2. The filename validation here is _not_ the same as done by paperclip,
3. The file contents have not yet been verified by paperclip.

If we change to `after_validation`, that method can assume that the filename is correct (base and extension), and the contents are correct as well.

The change(s) we'd make to the file base name will not invalidate the preceding paperclip validation.

Reply:
I see your point, after all I have the feeling this is not the right place to sanitize the file name since paperclip has many options for that, I would move this logic to a separate class and use the `filename_cleaner` option by paperclip https://github.com/thoughtbot/paperclip/blob/master/lib/paperclip/attachment.rb#L18, it is made exactly to cleanup invalid characters, meanwhile this should happen after the attachment validations.